Automattic/wp-calypso
# Modernize `date-range`

#### Changes proposed in this Pull Request

* Use `localized-moment` instead of `i18n-calypso`'s `moment` throughout
* Rewrite sub-components as functional components
* Add types to sub-components

#### Testing instructions

* Open `/devdocs/design/date-range`.
* Ensure that the component continues working correctly.
